packages/@react-aria/toast/src/useToast.ts

```diff
@@ -44,25 +47,47 @@ export function useToast(props: AriaToastProps): ToastAria {
       onAction(...args);
     }
 
-    if (shouldCloseOnAction && onClose) {
-      onClose(...args);
+    if (shouldCloseOnAction) {
+      onClose && onClose(...args);
+      onRemove && onRemove(toastKey);
     }
   };
 
   let iconProps = variant ? {alt: formatMessage(variant)} : {};
 
+  let pauseTimer = () => {
+    timer && timer.pause();
+  };
+
+  let resumeTimer = () => {
+    timer && timer.resume();
+  };
+
+  let {hoverProps} = useHover({
+    onHover: pauseTimer,
+    onHoverEnd: resumeTimer
+  });
+
+  let {focusProps} = useFocus({
+    onFocus: pauseTimer,
+    onBlur: resumeTimer
+  });
+
   return {
     toastProps: {
+      ...hoverProps,
       id: useId(id),
       role: 'alert'
     },
     iconProps,
     actionButtonProps: {
+      ...focusProps,
```

Comment:
I put the focus on the buttons for now, but it will probably be refactored when we do the focus followup work